Description: Add option to use system Poppler library
Author: mirabilos <tg@debian.org>
Forwarded: https://github.com/musescore/MuseScore/pull/4270
Applied-Upstream: master, commit:bcb75df

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -130,6 +130,7 @@ option(SOUNDFONT3    "Ogg Vorbis compres
 option(HAS_AUDIOFILE "Enable audio export" ON)                 # Requires libsndfile
 option(USE_SYSTEM_QTSINGLEAPPLICATION "Use system QtSingleApplication" OFF)
 option(USE_SYSTEM_FREETYPE "Use system FreeType" OFF)          # requires freetype >= 2.5.2, does not work on win
+option(USE_SYSTEM_POPPLER  "Use system Poppler for OMR" OFF)
 option(BUILD_LAME    "Enable MP3 export" ON)                   # Requires libmp3lame, call CMake with -DBUILD_LAME="OFF" to disable
 option(DOWNLOAD_SOUNDFONT "Download the latest soundfont version as part of the build process" ON)
 
@@ -785,7 +786,10 @@ if (ZERBERUS)
 endif (ZERBERUS)
 
 if (OMR)
-      subdirs (omr thirdparty/poppler)
+      subdirs (omr)
+      if (NOT USE_SYSTEM_POPPLER)
+            subdirs (thirdparty/poppler)
+      endif (NOT USE_SYSTEM_POPPLER)
 endif (OMR)
 
 if (OSC)
@@ -828,8 +832,13 @@ else (USE_SYSTEM_FREETYPE)
 endif (USE_SYSTEM_FREETYPE)
 
 if (OMR)
+ if (NOT USE_SYSTEM_POPPLER)
    include_directories(${PROJECT_SOURCE_DIR}/thirdparty/poppler)
    include_directories(${PROJECT_SOURCE_DIR}/thirdparty/poppler/poppler)
+ else (NOT USE_SYSTEM_POPPLER)
+   include_directories(/usr/include/poppler)
+   include_directories(/usr/include/poppler/qt5)
+ endif (NOT USE_SYSTEM_POPPLER)
 endif (OMR)
 
 add_subdirectory(mscore)
--- a/mscore/CMakeLists.txt
+++ b/mscore/CMakeLists.txt
@@ -608,7 +608,7 @@ if (MINGW)
       )
 
    if (OMR)
-      target_link_libraries(mscore omr poppler)
+      target_link_libraries(mscore omr poppler-qt5)
       if (OCR)
             target_link_libraries(mscore tesseract_api)
       endif (OCR)
@@ -756,7 +756,7 @@ else (MINGW)
          )
 
       if (OMR)
-         target_link_libraries(mscore omr poppler)
+         target_link_libraries(mscore omr poppler-qt5)
          if (OCR)
                target_link_libraries(mscore tesseract_api)
          endif (OCR)
@@ -869,7 +869,7 @@ else (MINGW)
          )
 
       if (OMR)
-         target_link_libraries(mscore omr poppler)
+         target_link_libraries(mscore omr poppler-qt5)
          if (OCR)
                target_link_libraries(mscore tesseract_api)
          endif (OCR)
--- a/mtest/cmake.inc
+++ b/mtest/cmake.inc
@@ -30,7 +30,7 @@ target_link_libraries(
       )
 
 if (OMR)
-      target_link_libraries(${TARGET} omr poppler)
+      target_link_libraries(${TARGET} omr poppler-qt5)
       if (OCR)
             target_link_libraries(${TARGET} tesseract_api)
       endif (OCR)
